akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -257,55 +314,157 @@ object TestSubscriber {
-      expectError(cause)
-      self
+      if (signalDemand) sub.request(1)
+      expectError()
     }
 
-    def expectSubscriptionAndError(): Throwable = {
+    /**
+     * Fluent DSL
+     *
+     * Expect subscription followed by immediate stream completion.
+     *
+     * By default `1` demand will be signalled in order to wake up a possibly lazy upstream.
+     *
+     * See also [[#expectSubscriptionAndComplete(Throwable, Boolean)]] if no demand should be signalled.
+     */
+    def expectSubscriptionAndError(cause: Throwable): Self =
+      expectSubscriptionAndError(cause, true)
+
+    /**
+     * Fluent DSL
+     *
+     * Expect subscription followed by immediate stream completion.
+     * By default `1` demand will be signalled in order to wake up a possibly lazy upstream
+     *
+     * See also [[#expectSubscriptionAndError(Throwable)]].
+     */
+    def expectSubscriptionAndError(cause: Throwable, signalDemand: Boolean): Self = {
       val sub = expectSubscription()
-      sub.request(1)
-      expectError()
+      if (signalDemand) sub.request(1)
+      expectError(cause)
+      self
     }
 
-    def expectSubscriptionAndComplete(): Self = {
+    /**
+     * Fluent DSL
+     *
+     * Expect subscription followed by immediate stream completion.
+     * By default `1` demand will be signalled in order to wake up a possibly lazy upstream
+     *
+     * See also [[#expectSubscriptionAndComplete(Boolean)]] if no demand should be signalled.
+     */
+    def expectSubscriptionAndComplete(): Self =
+      expectSubscriptionAndComplete(true)
+
+    /**
+     * Fluent DSL
+     *
+     * Expect subscription followed by immediate stream completion.
+     *
+     * Depending on the `signalDemand` parameter demand may be signalled immediatly after obtaining the subscription
+     * in order to wake up a possibly lazy upstream. You can disable this by setting the `signalDemand` parameter to `false`.
+     *
+     * See also [[#expectSubscriptionAndComplete]].
+     */
+    def expectSubscriptionAndComplete(signalDemand: Boolean): Self = {
       val sub = expectSubscription()
-      sub.request(1)
+      if (signalDemand) sub.request(1)
       expectComplete()
       self
     }
 
+    /**
+     * Fluent DSL
+     *
+     * Expect given next element or error signal, returning whichever was signalled.
+     */
+    def expectNextOrError(): Either[Throwable, I] = {
+      probe.fishForMessage(hint = s"OnNext(_) or error") {
+        case OnNext(element) ⇒ true
+        case OnError(cause)  ⇒ true
+      } match {
+        case OnNext(n: I @unchecked) ⇒ Right(n)
+        case OnError(err)            ⇒ Left(err)
+      }
+    }
+
+    /**
+     * Fluent DSL
+     * Expect given next element or error signal.
+     */
     def expectNextOrError(element: I, cause: Throwable): Either[Throwable, I] = {
       probe.fishForMessage(hint = s"OnNext($element) or ${cause.getClass.getName}") {
-        case OnNext(n)        ⇒ true
```

Comment:
the `n` instead of
```
`element` 
```

here is a bug - we'd accept any OnNext, not just the expected one.